common/web/gesture-recognizer/src/engine/headless/gestures/matchers/gestureMatcher.ts

```diff
@@ -450,13 +451,40 @@ export class GestureMatcher<Type, StateToken = any> implements PredecessorMatch<
         // pathMatcher for a source that failed to meet initial conditions.
         this.pathMatchers.pop();
 
-        this.finalize(false, 'path');
+      /*
+       * To prevent any further retries for the model (via rejectionActions), we list the
+       * cause as 'cancelled'.  The rejection-action mechanism already blocks paths that
+       * are rejected synchronously by this method; this adds an extra layer of protection
+       * and is likely also more clear.
+       *
+       * Alternatively, 'item' _should_ be fine - and corresponds best with a
+       * rejection based on the initial item.
+       */
```

Comment:
I don't really understand the second part of this comment. It's somewhat vague, using words like "likely" and "should" but it doesn't really clarify. Likely more clear than what? What part of the code uses the 'cancelled' cause and how does its behaviour differ? What is 'this method' -- is that 'this.finalize' or something else?

Saying that alternatively 'item' _should_ be fine suggests that this change is wrong... particularly given the second half of that sentence.

Code comments should not be describing the change that you've made (from 'path' to 'cancelled') -- that belongs in a commit comment. The code comment needs to stand on its own explaining what is happening and why.

Can we use consistent comment formatting? This is a nit but elsewhere you are using `//`. The `*` on the front of each line format should be reserved for JSDoc style comments.

Reply:
> Can we use consistent comment formatting? This is a nit but elsewhere you are using `//`. The `*` on the front of each line format should be reserved for JSDoc style comments.

I thought JSDoc style comments use a `/**` opening - two asterisks - rather than just `/*`?  Plain `/*` is just a plain multi-line comment, isn't it?  (VS Code certainly ignores `/*` when it comes to JSDoc - it only looks at the `/**` comments.)  

Yeah, in the past (and sometimes even currently) I've often used lots of `//` for larger multiline comments, but I've been trying to work on that in favor of using a plain multiline comment style for sizably multiline comments like this one.  Would it be better if I nixed the intermediate `*` line prefixes but kept it otherwise intact?

Part of why I've been trying to move away from `//` for larger comments like this is that they're a bit of a pain to maintain as line-width shifts; I find it less tedious to maintain that aspect when using a multi-line comment.

Reply:
Yeah, JSDoc needs the `/**` but then the subsequent lines look like JSDoc but aren't. But it's not a big deal, just a bit of a mental shift when moving between different comment blocks.

For formatting multi-line comments, I use the VSCode extension Rewrap -- select the comment, press Alt+Q and it will re-wrap it (or Alt+Shift+Q to unwrap). Also works wonders with Markdown, commit messages, etc etc. Saves me so much angst :grin: